### ISSUE TITLE
Support API to Organic Maps

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#14427]
### Tell us your idea!

As discussed in #13209 maps.me seems to go more and more away from opensource, changed its map style and even included in-app purchases (or similar).
As #13209 is closed as the existing API seems to work again, the idea of supporting Organic maps (developed open source based on a fork of maps.me) was disregarded.

### What solution would you suggest?

As we also get corresponding requests on support mail I would like to open this feature request to implement the API towards the Organic maps app.

See https://github.com/cgeo/cgeo/issues/13209#issuecomment-1382541572

There is an API available here:
https://github.com/organicmaps/api-android

To be discussed whether it shall replace the maps.me implementation or both exist in parallel.

**Comment by MakiWolf** [comment]
Why is maps.me listed in menu also on devices where it is not installed? Because osmand is not displayed on devices where it is not installed.
![grafik](https://github.com/cgeo/cgeo/assets/73300633/8bf2097e-0884-4ec8-8b8f-973ed9a15f59)
![grafik](https://github.com/cgeo/cgeo/assets/73300633/2a7521b7-476d-41dc-93f3-b7c424b54db0)
It would be interesting if there is much difference about the api use. Because organic maps is an fork from maps.me.

**Comment by moving-bits** [comment]
> Why is maps.me listed in menu also on devices where it is not installed?

This depends on whether c:geo is able to detect whether a certain app is installed or not. I guess we have no check implemented for it in this case, which could be either intentionally or due to an app using several different app id (not sure if that's the case with maps.me).

**Comment by Lineflyer** [comment]
It is listed because the API presents a link to install it in such cases.

**Comment by moving-bits** [comment]
See #14436 for a first stab at an OrganicMaps integration into c:geo

**Comment by MakiWolf** [comment]
Small sidenote has license of maps.me api also changed?
Source code reposositority? 
https://github.com/mapsme/api-android
Last updated in 2019

**Comment by ztNFny** [comment]
> Small sidenote has license of maps.me api also changed? Source code reposositority? https://github.com/mapsme/api-android Last updated in 2019

Yes, Maps.me has gone fully commercial, that's why OrganicMaps fork was created.

**Comment by MakiWolf** [comment]
> Yes, Maps.me has gone fully commercial, that's why OrganicMaps fork was created.

I would suggest to replace it with organic maps. cgeo should recommend opensource alternatives, when the same app is avilable in open source. But thats my opinion what do others think?

**Comment by moving-bits** [comment]
> > Yes, Maps.me has gone fully commercial, that's why OrganicMaps fork was created.
> 
> I would suggest to replace it with organic maps. cgeo should recommend opensource alternatives, when the same app is avilable in open source. But thats my opinion what do others think?

We should not remove an existing interface to a working counterpart, there may be users still using that app, maybe even having paid for the commercial version. No need to cripple c:geo for them. But we could (and IMHO should) remove the automatic download offer for maps.me due to them no longer being open source.

Regarding the "recommendations" part: The OrganicMaps implementation with PR #14436 will not only support OrganicMaps but also offer download of the app if it's not yet available on the user's system.

**Comment by MakiWolf** [comment]
@moving-bits Just found out that when open the cache in organic maps from cgeo, then switching to start screen and open cgeo app (without completly closing cgeo), cgeo shows organic maps application. Is that a bug or feature from the api?

**Comment by moving-bits** [comment]
@MakiWolf 
c:geo calls OrganicMaps by using an `Intent`, which makes the called `Activity` sort of being part of the calling app. Same as with other places where c:geo integrates / calls external building blocks.

**Comment by MakiWolf** [comment]
> Regarding the "recommendations" part: The OrganicMaps implementation with PR #14436 will not only support OrganicMaps but also offer download of the app if it's not yet available on the user's system.

@moving-bits For interesting. Why are some apps also listed in utility programs and some apps that are supported to open in the app from cgeo are not in this list there? Like brouter where cgeo can also use the external app as routing engine?
![grafik](https://github.com/cgeo/cgeo/assets/73300633/b8210384-9706-4dd1-996a-ae0a28760b3e)
![grafik](https://github.com/cgeo/cgeo/assets/73300633/0a4fbe45-623e-4307-a3c9-103d213a7622)

**Comment by moving-bits** [comment]
Just stumbled upon this remark - sorry for replying so late...

> [@moving-bits](https://github.com/moving-bits) For interesting. Why are some apps also listed in utility programs and some apps that are supported to open in the app from cgeo are not in this list there? Like brouter where cgeo can also use the external app as routing engine?

Adding to or removing from "useful apps" list happens on a per-entry basis, there's no specific process for it. BRouter got removed from this list, as you don't have to install it anymore, as c:geo uses an internal implementation. And if you don't activate the internal implementation, it will automatically find the external one.

**Comment by moving-bits** [comment]
We have a working basic integration to Organic Maps meanwhile, so let's close here. If any specific new features shall be implemented, separate issues should be opened.
